internal/service/distribution/distribution.go

```diff
@@ -0,0 +1,172 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package distribution
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+
+	containerdimages "github.com/containerd/containerd/images"
+	dockerresolver "github.com/containerd/containerd/remotes/docker"
+	cremoteerrors "github.com/containerd/containerd/remotes/errors"
+	cerrdefs "github.com/containerd/errdefs"
+	"github.com/containerd/nerdctl/pkg/imgutil/dockerconfigresolver"
+	dockertypes "github.com/docker/cli/cli/config/types"
+	registrytypes "github.com/docker/docker/api/types/registry"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+
+	"github.com/runfinch/finch-daemon/api/handlers/distribution"
+	"github.com/runfinch/finch-daemon/internal/backend"
+	"github.com/runfinch/finch-daemon/pkg/errdefs"
+	"github.com/runfinch/finch-daemon/pkg/flog"
+	"github.com/runfinch/finch-daemon/pkg/utility/authutility"
+	"github.com/runfinch/finch-daemon/pkg/utility/imageutility"
+)
+
+type service struct {
+	client       backend.ContainerdClient
+	nctlImageSvc backend.NerdctlImageSvc
+	logger       flog.Logger
+}
+
+// setting getAuthCredsFunc as a variable to allow mocking this function for unit testing.
+var getAuthCredsFunc = authutility.GetAuthCreds
+
+func NewService(client backend.ContainerdClient, nerdctlImageSvc backend.NerdctlImageSvc, logger flog.Logger) distribution.Service {
+	return &service{
+		client:       client,
+		nctlImageSvc: nerdctlImageSvc,
+		logger:       logger,
+	}
+}
+
+func (s *service) Inspect(ctx context.Context, name string, ac *dockertypes.AuthConfig) (*registrytypes.DistributionInspect, error) {
```

Comment:
nit: function can be broken down for the different operation its doing, larger function are usually difficult to unit test.